libraries/BLE/src/BLECallbacks.cpp

```diff
@@ -0,0 +1,205 @@
+        pvalue = blecharacteritic->value();
+        return bt_gatt_attr_read(conn, attr, buf, len, offset, pvalue,
+                                 blecharacteritic->valueLength());
+    }
+    else if (BLETypeDescriptor == type)
+    {
+        BLEDescriptorImp* bledescriptor = (BLEDescriptorImp*)bleattr;
+        pvalue = bledescriptor->value();
+        return bt_gatt_attr_read(conn, attr, buf, len, offset, pvalue, bledescriptor->valueLength());
+    }
+    return 0;
+}
+
+// GATT server only
+ssize_t profile_write_process(bt_conn_t *conn,
+                              const bt_gatt_attr_t *attr,
+                              const void *buf, uint16_t len,
+                              uint16_t offset)
+{
+    pr_info(LOG_MODULE_BLE, "%s1", __FUNCTION__);
+    BLEAttribute *bleattr = (BLEAttribute *)attr->user_data;
+    BLECharacteristicImp* blecharacteritic;
+    BLEAttributeType type = bleattr->type();
+    if ((BLETypeCharacteristic != type) || 0 != offset)
+    {
+        return 0;
+    }
+    
+    blecharacteritic = (BLECharacteristicImp*)bleattr;
+    blecharacteritic->setValue((const uint8_t *) buf, len);
+    return len;
+}
+
+ssize_t profile_longwrite_process(struct bt_conn *conn,
+                                     const struct bt_gatt_attr *attr,
+                                     const void *buf, uint16_t len,
+                                     uint16_t offset)
+{
+    BLECharacteristicImp *blecharacteritic = (BLECharacteristicImp*)attr->user_data;
+    
+    blecharacteritic->setBuffer((const uint8_t *) buf, len, offset);
+    
+    return len;
+}
+
+int profile_longflush_process(struct bt_conn *conn,
+                              const struct bt_gatt_attr *attr, 
+                              uint8_t flags)
+{
+    BLECharacteristicImp *blecharacteritic = (BLECharacteristicImp*)attr->user_data;
+
+    switch (flags)
+    {
+    case BT_GATT_FLUSH_DISCARD:
+        /* Discard buffer reseting it back with data */
+        blecharacteritic->discardBuffer();
+        return 0;
+    case BT_GATT_FLUSH_SYNC:
+        /* Sync buffer to data */
+        blecharacteritic->syncupBuffer2Value();
+        return 0;
+    }
+
+    return -EINVAL;
+}
+
+
+// GATT client only
+uint8_t profile_notify_process (bt_conn_t *conn,
+                                bt_gatt_subscribe_params_t *params,
+                                const void *data, uint16_t length)
+{
+    //BLEPeripheralHelper* peripheral = BLECentralRole::instance()->peripheral(conn);// Find peripheral by bt_conn
+    //BLEAttribute* notifyatt = peripheral->attribute(params); // Find attribute by params
+    BLECharacteristicImp* chrc = NULL;
+    BLEDevice bleDevice(bt_conn_get_dst(conn));
+    BLEProfileManager::instance()->characteristic(bleDevice, params->value_handle);
+    
+    //assert(notifyatt->type() == BLETypeCharacteristic);
+    pr_debug(LOG_MODULE_APP, "%s1", __FUNCTION__);
+    
+    chrc->setValue((const unsigned char *)data, length);
```

Comment:
It looks to me like `chrc` will *always* be NULL... it is set to NULL on line 98, and then it is not modified again. However you attempt to dereference it here, on line 105, which will lead to a system crash.